include/zarray/zarray_impl.hpp

```diff
@@ -220,6 +335,7 @@ namespace xt
         zarray_wrapper(const zarray_wrapper&) = default;
 
         CTE m_array;
+        mutable nlohmann::json m_attrs;
```

Comment:
Why mutable? It seems that this object is always returned by const reference.

Reply:
Without `mutable` we get this kind of errors:
```
In file included from /home/david/github/davidbrochart/zarray/include/zarray/zarray.hpp:19,
                 from /home/david/github/davidbrochart/zarray/test/test_zarray.cpp:11:
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp: In instantiation of ‘const json& xt::zarray_wrapper<CTE>::attrs() const [with CTE = const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&; nlohmann::json = nlohmann::basic_json<>]’:
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:690:17:   required from here
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:692:25: error: no matching function for call to ‘get_attrs(const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&, const json&)’
  692 |         return get_attrs(m_array, m_attrs);
      |                ~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:42:5: note: candidate: ‘template<class T> typename std::enable_if<xt::HasAttrs<T>::value, const nlohmann::basic_json<>&>::type xt::get_attrs(T&, nlohmann::json&)’
   42 |     get_attrs(T& t, nlohmann::json& attrs)
      |     ^~~~~~~~~
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:42:5: note:   template argument deduction/substitution failed:
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:692:35: note:   cannot convert ‘((const xt::zarray_wrapper<const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&>*)this)->xt::zarray_wrapper<const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&>::m_attrs’ (type ‘const json’ {aka ‘const nlohmann::basic_json<>’}) to type ‘nlohmann::json&’ {aka ‘nlohmann::basic_json<>&’}
  692 |         return get_attrs(m_array, m_attrs);
      |                                   ^~~~~~~
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:49:5: note: candidate: ‘template<class T> typename std::enable_if<(! xt::HasAttrs<T>::value), const nlohmann::basic_json<>&>::type xt::get_attrs(T&, nlohmann::json&)’
   49 |     get_attrs(T& t, nlohmann::json& attrs)
      |     ^~~~~~~~~
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:49:5: note:   template argument deduction/substitution failed:
/home/david/github/davidbrochart/zarray/include/zarray/zarray_impl.hpp:692:35: note:   cannot convert ‘((const xt::zarray_wrapper<const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&>*)this)->xt::zarray_wrapper<const xt::xarray_container<xt::uvector<int, std::allocator<int> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>&>::m_attrs’ (type ‘const json’ {aka ‘const nlohmann::basic_json<>’}) to type ‘nlohmann::json&’ {aka ‘nlohmann::basic_json<>&’}
  692 |         return get_attrs(m_array, m_attrs);
      |
```

Reply:
Then `get_attrs` should take its second argument by const ref

Reply:
That was it, thanks!